### PR TITLE
Moves NuGet 6.2 shipped public APIs to PublicAPI.Shipped.txt files (dev branch)

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1617

Regression? Last working version: N/A

## Description

Moves all public APIs shipped in 6.2 to Public.API.Shipped.txt files

This PR cherry-picks commit from https://github.com/NuGet/NuGet.Client/pull/4621 and adds additional commits to get a green CI build.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A: Green CI build is the test

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A: No product changes.

### CODE REVIEW

**Review by zivkan — DISMISSED** [review]

**Review by jeffkl — DISMISSED** [review]

**Comment by dominoFire** [comment]
Verified that this PR has the same number of files changed compared to PR to release-6.2.x branch.

PublicAPI.*.txt files are different on each branch.

I will merge once I get a green light with rebased latest dev branch.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**
